[flare] Add cloud-run flare documentation & update lambda flare documentation (#1037)
DataDog/datadog-ci: src/commands/cloud-run/README.md

> ## Troubleshooting Cloud Run Instrumentation

Reviewer: Missing `Configuration` section for credentials, env variables for site, api key, etc.

> To troubleshoot issues you may be encountering with Datadog monitoring on your Cloud Run services, use the `datadog-ci cloud-run flare` command in the root of your project directory. This command collects important data about a Cloud Run service, such as environment variables and configuration information. These files will be submitted to Datadog support via a ticket matching the provided Zendesk case ID.

Reviewer 2: ```suggestion
To troubleshoot issues you encounter with Datadog monitoring on your Cloud Run services, run the `datadog-ci cloud-run flare` command in the root of your project directory. This command collects important data about the Cloud Run service, such as environment variables and configuration information. These files will be submitted to Datadog support via a ticket matching the provided Zendesk case ID.
```

**Examples**
```bash
# Collect and send files to Datadog support for a single service
datadog-ci cloud-run -s <service> -p <project> -r <region/location> -c <case-id> -e <email-on-case-id>

# Include recent logs
datadog-ci cloud-run -s <service> -p <project> -r <region/location> -c <case-id> -e <email-on-case-id> --with-logs

# Dry run: collect data, but don't send to Datadog support
datadog-ci cloud-run -s <service> -p <project> -r <region/location> -c <case-id> -e <email-on-case-id> --with-logs --dry-run
```

**Arguments**

| Argument              | Shorthand | Description                                                                                                                 | Default |
|-----------------------|-----------|-----------------------------------------------------------------------------------------------------------------------------|---------|
| `--service`           | `-s`      | The name of the Cloud Run service.                                                                                          |         |

> | `--project`           | `-p`      | The name of the Google Cloud project within which the Cloud Run service is hosted.                                          |         |

Reviewer 2: ```suggestion
| `--project`           | `-p`      | The name of the Google Cloud project where the Cloud Run service is hosted.                                          |         |
```

| `--region`            | `-r`      | The region where the Cloud Run service is hosted.                                                                           |         |
| `--case-id`           | `-c`      | The Datadog case ID to send the files to.                                                                                   |         |
| `--email`             | `-e`      | The email associated with the specified case ID.                                                                            |         |
| `--with-logs`         |           | Collect recent logs for the specified service.                                                                              | `false` |

> | `--start` and `--end` |           | Define a time range in milliseconds since the Unix Epoch to gather logs within that range. (`--with-logs` must be included) |         |

Reviewer 2: ```suggestion
| `--start` and `--end` |           | Define a time range in milliseconds since the Unix Epoch to gather logs within that range. (`--with-logs` must be included.) |         |
```

Reviewer 2: I'm getting slightly tripped up by the wording "since the Unix Epoch to gather". Can you clarify what that means?

Author: "Milliseconds since Unix Epoch" represents a timestamp. The user can specify two times:`--start` and `--end` to only get logs within that time range. I have updated the wording a bit to be more clear, WDYT?

> | `--dry-run`           | `-d`      | Preview collected data which would be sent to Datadog support.                                                              | `false` |

Reviewer 2: ```suggestion
| `--dry-run`           | `-d`      | Preview data that will be sent to Datadog support.                                                              | `false` |
```

## Community

For product feedback and questions, join the `#serverless` channel in the [Datadog community on Slack](https://chat.datadoghq.com/).

> [1]: https://cloud.google.com/sdk/gcloud/reference/auth/login

Reviewer: Where are these links even being used?

Author: They're not directly being referenced. But I noticed that some links in `lambda/README.md` aren't directly used either. For example, the second 
`[2]: https://github.com/DataDog/datadog-ci`
is not referenced in lambda's README

Reviewer: Well, if we are not referencing any of these, and they don't appear in the docs, why are we adding them? Same goes for Lambda then.

Author: Ah I didn't realize that they are not shown at the bottom in Markdown. I thought these were just relevant links. I can remove the unused ones here and on Lambda

[2]: https://github.com/DataDog/datadog-ci
[3]: https://docs.datadoghq.com/serverless/google_cloud_run

> [4]: https://docs.datadoghq.com/account_management/api-app-keys/#api-keys

Author: Any more relevant links I'm missing?

## Configuration

### GCP Credentials

You must have valid [GCP credentials][1] configured with access to the Lambda and CloudWatch services where you are running any `datadog-ci lambda` command. You can configure credentials by running `gcloud auth application-default login` and following the prompts in your browser. 

### Environment variables

Expose these environment variables in the environment where you are running `datadog-ci cloud-run flare`:

| Environment Variable | Description                                                                                                                                                                                                                                      | Example                               |
|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------|

> | `DATADOG_API_KEY`    | Datadog API Key. Used to attach the flare files to your Zendesk ticket. For more information about getting a Datadog API key, see the [API key documentation][2].                                                                                | `export DATADOG_API_KEY=<API_KEY>`    |

Reviewer: Should we change this to `DD_API_KEY` and `DD_SITE` since we already support those?
